app/ui/src/app/integration/edit-page/flow-view/flow-view.component.ts

```diff
@@ -81,39 +84,39 @@ export class FlowViewComponent extends ChildAwarePage
   }
 
   loaded() {
-    return this.currentFlow.loaded;
+    return this.currentFlowService.loaded;
   }
 
   get currentStep() {
-    return this.getCurrentStep();
+    return this.flowPageService.getCurrentStep(this.route);
   }
 
   startConnection() {
-    return this.currentFlow.getStartStep();
+    return this.currentFlowService.getStartStep();
   }
 
   endConnection() {
-    return this.currentFlow.getEndStep();
+    return this.currentFlowService.getEndStep();
   }
 
   firstPosition() {
-    return this.currentFlow.getFirstPosition();
+    return this.currentFlowService.getFirstPosition();
   }
 
   lastPosition() {
-    return this.currentFlow.getLastPosition();
+    return this.currentFlowService.getLastPosition();
   }
 
   getMiddleSteps() {
-    return this.currentFlow.getMiddleSteps();
+    return this.currentFlowService.getMiddleSteps();
```

Comment:
We can leave these getters as they are ATM, but in the future we should get rid of getter members that execute functions (service methods here) internally, since those will be evaluated as often as template bindings.

A good way to address this is to refactor all those service methods to return Observable streams, so we can bind assign them to observable properties in the class and bind these directly in the template along with the `async` pipe.

Reply:
Good call!  I'll plan on that for future work...